modules/postgresql/src/test/java/org/testcontainers/containers/PostgreSQLConnectionURLTest.java

```diff
@@ -0,0 +1,25 @@
+package org.testcontainers.containers;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.assertEquals;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertFalse;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertTrue;
+import static org.testcontainers.jdbc.ConnectionUrl.newInstance;
+
+public class PostgreSQLConnectionURLTest {
```

Comment:
Can we please have a unit test for the "corrupted JDBC URLs" issue?
I believe we don't actually need to start any containers to verify it.

Reply:
The change was TDD-ed - if you check out https://github.com/testcontainers/testcontainers-java/commit/8803fe1f0a389c17be0f8bc6a3c5ed0a18e3de55, you can see that the test fails by producing illegal query string `...?loggerLevel=OFF?stringtype=unspecified&stringtype=unspecified` (`?` after `loggerLevel=OFF` instead of `&`).

And if you check out HEAD, you can see that it's generated correctly (and the test passes):
`...?loggerLevel=OFF&stringtype=unspecified&stringtype=unspecified`

...or did you mean a name change?

And nope, the method responsible for generating the connection URL depends on the container's IP and port. I have a few ideas on how to bypass the problem, but none of them are acceptable, I think

Reply:
The test contains an integration test, while it can be unit tested.

Consider writing a unit test instead, so that you don't start a container at all to test the string concatenation

Reply:
Sure, here you go: https://github.com/testcontainers/testcontainers-java/pull/2049/commits/50e75fa09780b5762d27ffece7f585073dc24a5e

The trade-off is that now we're not testing the real thing so there's a possibility that the real thing gets broken and tests will still pass.

Reply:
@pivovarit please do not resolve conversations that were not started by you.